src/eq_schema/Block.js

```diff
@@ -8,16 +9,63 @@ const pageTypeMappings = {
 };
 
 class Block {
-  constructor(title, description, page) {
+  constructor(title, description, page, ctx) {
     this.id = `block${page.id}`;
     this.title = getInnerHTML(title);
     this.description = getInnerHTML(description);
     this.type = this.convertPageType(page.pageType);
-    this.questions = this.buildQuestions(page);
+    this.questions = this.buildQuestions(page, ctx);
+    if (!isNil(page.routingRuleSet)) {
+      // eslint-disable-next-line camelcase
+      this.routing_rules = this.buildRoutingRules(
+        page.routingRuleSet.routingRules,
+        page.routingRuleSet,
+        page.id,
+        ctx
+      );
+    }
   }
 
-  buildQuestions(page) {
-    return [new Question(page)];
+  buildQuestions(page, ctx) {
+    return [new Question(page, ctx)];
+  }
+
+  buildRoutingRules(routingRules, routingRuleSet, pageId, ctx) {
+    const flattenedRules = flatMap(routingRules, rule => {
+      if (rule.operation === "And") {
+        return rule;
+      }
+      return flatMap(rule.conditions, (cond, i) => {
+        return cond.routingValue.value.map(answerValue => {
+          return {
+            id: `${rule.id}-${i}`,
+            operation: "And",
+            conditions: [
+              {
+                id: i,
+                comparator: cond.comparator,
+                answer: cond.answer,
+                routingValue: {
+                  value: [answerValue]
+                }
+              }
+            ],
+            goto: rule.goto
+          };
+        });
+      });
+    });
+
+    const elseRule = function(elseRule) {
+      return {
+        goto: elseRule.else
```

Comment:
Else refers to a QuestionPage or a Section here?

Reply:
Probably worth naming `buildElse` for consistency and in light of changes we've discussed this could return RoutingDestination(else)?